src/libsync/clientstatusreporting.cpp

```diff
@@ -0,0 +1,440 @@
+/*
+ * Copyright (C) 2023 by Oleksandr Zolotov <alex@nextcloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+#include "clientstatusreporting.h"
+
+#include "account.h"
+#include "clientstatusreportingrecord.h"
+#include <configfile.h>
+#include "common/c_jhash.h"
+#include <networkjobs.h>
+
+namespace
+{
+constexpr auto lastSentReportTimestamp = "lastClientStatusReportSentTime";
+constexpr auto statusNamesHash = "statusNamesHash";
+
+constexpr auto statusReportCategoryE2eErrors = "e2e_errors";
+constexpr auto statusReportCategoryProblems = "problems";
+constexpr auto statusReportCategorySyncConflicts = "sync_conflicts";
+constexpr auto statusReportCategoryVirus = "virus_detected";
+}
+
+namespace OCC
+{
+Q_LOGGING_CATEGORY(lcClientStatusReporting, "nextcloud.sync.clientstatusreporting", QtInfoMsg)
+
+ClientStatusReporting::ClientStatusReporting(Account *account, QObject *parent)
```

Comment:
@mgallien `ClientStatusReporting` and related classes must not take ownership of the `Account` instance. This will prevent deinitialization and might introduce other issues. `Account` owns a unique instance of `ClientStatusReporting` and it destroys it when it is being destroyed itself. Then `ClientStatusReporting` gets destroyed and its database and networking class is destroyed as well. What you suggest would introduce a circular dependency on the lifetime.